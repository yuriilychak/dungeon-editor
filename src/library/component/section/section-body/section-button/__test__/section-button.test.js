import React from "react";
import {createMount} from "@material-ui/core/test-utils";

import SectionButton from "../section-button";

describe("SectionButton test", () => {
    const onClick = jest.fn();
    const Icon = () => <div>Icon</div>;
    const wrapper = createMount()(
        <SectionButton
            title="test"
            Icon={Icon}
            onClick={onClick}
            fileId={5}
            sectionId={3}
            userData="test"
        />
    );

    it("default snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("test click", () => {
        wrapper.find("button").simulate("click");
        expect(onClick).toHaveBeenCalled();
    });
});
