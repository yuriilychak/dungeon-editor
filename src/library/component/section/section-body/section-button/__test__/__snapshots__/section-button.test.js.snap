// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionButton test default snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-haspopup=\\"true\\" title=\\"test\\"><span class=\\"MuiIconButton-label\\"><div>Icon</div></span><span class=\\"MuiTouchRipple-root\\"></span></button>"`;
